src/shared/utils/__tests__/logging.spec.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-console */
+import { createGtagEvent } from '../logging';
+
+describe('createGtagEvent', () => {
+  it('should deal with objects', () => {
+    expect(createGtagEvent('log', { foo: 'bar', baz: ['baz', 'qux'] })).toEqual(
+      {
+        event_baz: '["baz","qux"]',
+        event_foo: '"bar"',
+        event_label: 'log',
+      }
+    );
+  });
+  it('should deal with strings', () => {
+    expect(createGtagEvent('log', 'foo')).toEqual({
+      event_data: 'foo',
+      event_label: 'log',
+    });
+  });
+  it('should deal with numbers', () => {
+    expect(createGtagEvent('log', 12345)).toEqual({
+      event_data: 12345,
+      event_label: 'log',
+    });
+  });
+
+  it('should deal with booleans', () => {
+    expect(createGtagEvent('log', true)).toEqual({
+      event_data: true,
+      event_label: 'log',
+    });
+  });
+
+  it('should warn when event data type is not handled', () => {
+    console.warn = jest.fn();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (global as any).LIVE_RELOAD = false;
```

Comment:
Wouldn't you be able to use `window` since it's using js-dom, and the corresponding types in `webpack-define-plugin.d.ts` would be automatically applied here?

Reply:
Or maybe use the new `globalThis`, but not sure if it is supported in Typescript yet (it is in Node since v12 though)